package.xml

```diff
@@ -13,6 +13,7 @@
   <depend>ros2cli</depend>
 
   <exec_depend>openssl</exec_depend>
+  <exec_depend>python3-lxml</exec_depend>
```

Comment:
AFAICT, there is not rosdep rule for `python3-lxml`:

```
sros2: Cannot locate rosdep definition for [python3-lxml]
```

Reply:
I'm not sure how to add a package key to the ros2/rosdistro listing as one would normally do for ros1, as there doesn't seem to be a `rosdep/python.yaml` in the ros2 fork given the structure has changed. How would I go about doing the ros2 equivalent of this: https://github.com/ros/rosdistro/pull/12314

python3-lxml is already released in the official ubuntu, debian, and fedora distros:  
* https://packages.ubuntu.com/bionic/python3-lxml
* https://packages.debian.org/stretch/python3-lxml
* https://apps.fedoraproject.org/packages/python3-lxml

Reply:
We're still using the ROS 1 repo for rosdep: https://github.com/ros/rosdistro/tree/master/rosdep
You can add the key there.

Reply:
@jacobperron , please see https://github.com/ros/rosdistro/pull/19824

Reply:
Does `python3-lxml` also need to be a `<test_depend>` or just a `<depend>`?

Reply:
No, I don't think so. According to http://wiki.ros.org/catkin/package.xml#Dependencies:

> They should never duplicate any dependencies already mentioned as build or run dependencies.

Reply:
Can the built in `xml` module be used instead of `lxml`? https://docs.python.org/3.6/library/xml.etree.elementtree.html

Reply:
> Can the built in `xml` module be used instead of `lxml`?

If all we needed was to parse XML markup, then yes the standard python module `xml` would suffice. However in the interest of keeping the templating pipeline implementation agnostic (so that SROS2 security profiles may be consumed by other tools and languages), I've elected to use XML templating as well. This necessitates the use of both schema validation and template processing for the complete profile-to-permission pipeline. There are few python modules that provide XSD validation, and fewer still that also support XST templating. At the moment, `lxml` seems to be the only game in town.

https://stackoverflow.com/questions/299588/validating-with-an-xml-schema-in-python

For a while with a previous work, I was using `xmlschema`; a python native XSD engine (lacks a XST processor). Yet I was somewhat misusing it as a XST engine as well for pipelining permission translation; something a that proper XST files are much more adapt as solving.

https://pypi.org/project/xmlschema/

Also, I'm not sure if the latest fix to the standard modul's `Xincludes` processor as been backported to the version of python we target:

https://bugs.python.org/issue20928 

Thus the case for `lxml`.